### PR TITLE
Validation for request DTOs

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Services/AssignmentTaskService.cs [hunk 1]
```diff
@@ -19,27 +20,24 @@ public class AssignmentTaskService : IAssignmentTaskService
...
-
         var entity = _mapper.Map<AssignmentTask>(dto);
 
-        if (await _repositoryAssignment.FirstOrDefaultAsync(assignment => assignment.Id == dto.AssignmentId) == null)
+        var assignment = await _assignmentRepository.FirstOrDefaultAsync(assignment => assignment.Id == dto.AssignmentId);
+
+        if (assignment == null)
         {
-            throw new Exception($"This assingment not in database with this FK: {dto.AssignmentId}");
+            throw new AssignmentNotFoundException($"Assignment not found in the database with this ID: {dto.AssignmentId}");
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AssignmentTaskService.cs:37]: Better to create NotFoundException to use it elsewhere

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 2]
```diff
@@ -27,12 +28,12 @@ public async Task<AuthSuccessDTO> LoginAsync(LoginUserDTO user)
...
 
         if (existingUser == null)
-            throw new KeyNotFoundException(user.Email);
+            throw new UserNotFoundException($"Unable to find user by specified email. Email: {user.Email}");
 
         var result = await _userManager.CheckPasswordAsync(existingUser, user.Password);
 
         if(!result)
-            throw new UnauthorizedAccessException(user.Email);
+            throw new UserUnauthorizationException($"User input incorrect password. Password: {user.Password}");
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:36]: InvalidCredentialsException sounds better, I guess

File: StudyHub/StudyHub/Controllers/AssignmentTaskController.cs [hunk 3]
```diff
@@ -9,23 +11,22 @@ namespace StudyHub.Controllers;
...
-        try
-        {
-            var result = await _serviceTask.AddTask(dto);
-            return Ok(result);
-        }
-        catch (Exception ex)
-        {
-            return BadRequest(ex.Message);
-        }
+        _optionsValidator.ValidateAndThrow(dto.Options);
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Controllers/AssignmentTaskController.cs:27]: Can't you validate whole **dto** here, including **dto.Options**?
Seems logically for me

Comment (aquaprogit) [reply StudyHub/StudyHub/Controllers/AssignmentTaskController.cs:27]: Still not resolved :)

**Review by Podgore — APPROVED** [review]

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — APPROVED** [review]

**Pull request merged**